torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -79,7 +76,7 @@ struct CpuChannelRegistration {
 C10_DECLARE_REGISTRY(TensorPipeCpuChannelRegistry, CpuChannelRegistration);
 
 struct CudaChannelRegistration {
```

Comment:
Would it be possible to define a template `ChannelRegistration<TBuffer>` only once and then instantiate it twice with the two buffer types? I remember you said that registries don't like templates, is it this what you were referring to?

Reply:
This will run into the same issue of introducing more ifdefs as `ChannelRegistration<CudaContext>` won't exist if CUDA is not available. And I recall I tried this in the first attempt, but `C10_DECLARE_REGISTRY` doesn't seem to like it. Didn't spend too much time to dig.